File: ts/project/.sdk/tm/c/utility/result_body.c
// result_body utility (mirrors utility/result_body.rs).

#include "sdk.h"

SdkResult* result_body_util(Context* ctx) {
  Response* response = ctx->response;
  SdkResult* result = ctx->result;

  if (result && response) {
    voxgig_value* json = response->json;
    voxgig_value* body = response->body;
    if (voxgig_is_func(json) && !v_is_noval(body) && !v_is_null(body)) {
      result->body = call_json(json);
    }
  }
  return result;
}

File: ts/project/.sdk/tm/rust/utility/result_body.rs
use std::cell::RefCell;
use std::rc::Rc;

use crate::core::context::Context;
use crate::core::helpers::call_json;
use crate::core::result::SdkResult;
use crate::utility::voxgigstruct::Value;

pub fn result_body_util(ctx: &Rc<Context>) -> Option<Rc<RefCell<SdkResult>>> {
    let response = ctx.response.borrow().clone();
    let result = ctx.result.borrow().clone();

    if let Some(result) = &result {
        if let Some(response) = &response {
            let (json, body) = {
                let r = response.borrow();
                (r.json.clone(), r.body.clone())
            };
            if matches!(json, Value::Func(_)) && !body.is_noval() && !body.is_null() {
                result.borrow_mut().body = call_json(&json);
            }
        }
    }

    result
}
